abci/types/types.proto

```diff
@@ -207,8 +207,8 @@ message ConsensusParams {
 // BlockSize contains limits on the block size.
 message BlockSize {
   int32 max_bytes = 1;
-  int32 max_txs = 2;
-  int64 max_gas = 3;
+  int64 max_gas = 2;
+  int32 max_num_evidences = 3;
```

Comment:
do we need to expose this in here? maybe we can set some internal heuristic and let all just be bounded by `max_bytes` ?

Reply:
What do you mean by this? Don't we want this to be a consensus parameter, and since it pertains to parts of a block size it belongs in this object?

If I understand correctly, your suggesting we have a function inside of the code that calculates `max_num_evidence = f(max_bytes)`. This seems harder to configure, and as a user of the protocol I would want an easy way to alter this function. (At which point it would then have to go back into the config)

Reply:
>  This seems harder to configure, and as a user of the protocol I would want an easy way to alter this function.

Do you really though? Seems like it should basically be as large as possible - if there's evidence, Tendermint should prioritize including it. Not clear that it's really something that needs to be exposed to users right now.